### PR TITLE
Added the new prometheus-thanos chart #D2IQ-64966

### PR DESCRIPTION
**Pull request by aayvazyan**
### What this PR does / why we need it:
This is an umbrella chart made from the current [prometheus-operator](https://github.com/mesosphere/charts/tree/master/staging/prometheus-operator) plus [Thanos](https://github.com/thanos-io/thanos). 

The aim of this chart is to have the full Thanos functionality within the [`prometheus`](https://github.com/mesosphere/kubernetes-base-addons/tree/master/addons/prometheus/0.35.x) baseaddon and to be able to configure Thanos for any Konvoy cluster from within the `cluster.yaml`  

The KEP is under review at [`mesosphere/ksphere-platform:sre/prometheus-thanos-kep`](https://github.com/mesosphere/ksphere-platform/blob/sre/prometheus-thanos-kep/keps/sig-ksphere-observability/20200229-kep-prometheus-thanos.md)

Ticket: https://jira.d2iq.com/browse/D2IQ-64973

Epic: https://jira.d2iq.com/browse/D2IQ-64981

Scope of Work Doc: [Thanos Scope of Work](https://docs.google.com/document/d/1S5APcDXRDzrrTgoGf09Bcr1onbXZu2N-fMVfRgzEBqU/edit)

### CODE REVIEW

File: staging/prometheus-thanos/crds/crd-alertmanager.yaml [hunk 1]
```diff
@@ -0,0 +1,4481 @@
+# https://raw.githubusercontent.com/coreos/prometheus-operator/master/example/prometheus-operator-crd/monitoring.coreos.com_alertmanagers.yaml
```
Comment (joejulian) [inline staging/prometheus-thanos/crds/crd-alertmanager.yaml]: Please follow these guidelines for CRDs. Yes, it's only documented on my blog. I'll pull that content into a document in the charts repo, but for now: https://joejulian.name/post/helm-v2-crd-management/

**Comment by alejandroEsc** [comment]
> I think we should pull the current mesosphere `prometheus-operator` (our fork) as a subchart to the parent `prometheus-thanos` chart. So essentially just copying the chart entirely into a `staging/prometheus-thanos/charts/prometheus-operator/` subchart dir. It looks like with these changes, the `prometheus-operator` chart was actually just renamed to `prometheus-thanos` and enhanced to pull `thanos` chart in. But we really want to be customizing our fork of `prometheus-operator` as little as possible - it is still our plan in the future to eventually get rid of our fork entirely and use the official upstream helm chart.
> 
> On that note, we're trying to stay in line with the upstream chart as closely as possible to make that transition smooth. We shouldn't update the subcharts within prometheus-operator since we're still operating off the upstream chart as our base, and each version of the chart has been tested and vetted with each version of the subcharts within it. If we update those subcharts, we can't ensure that those versions are compatible with the version of the prometheus-operator chart.

Not directly copying, please use `helm dependency` tooling for this.

**Comment by alejandroEsc** [comment]
I am not sure I follow the thought here. @gracedo @branden could we please discuss in zoom sometime? This will have some effect, including on KBA and our releases, I want to make sure that we are on the same page here.

**Comment by alejandroEsc** [comment]
@aayvazyan awesome description! thanks for those details.

**Comment by shaneutt** [comment]
This is still marked as `WIP`, is that accurate? Is any assistance needed to move this one forward?

File: staging/prometheus-thanos/crds/crd-alertmanager.yaml (hunk 1 above)
Comment (aayvazyan) [reply staging/prometheus-thanos/crds/crd-alertmanager.yaml]: @joejulian can you please be more specific what changes you request to do? To my understanding what you request is already there. https://github.com/mesosphere/charts/commit/4982c96dd316f2f93cefd9eab61b67f4205a0a95

**Comment by aayvazyan** [comment]
@shaneutt We are testing the suggested scenario with having prometheus-operator as a subchart instead of copying it. We'll close or get back to this PR as soon as all the suggested scenarios are tested and we have a final decision on how we go on with Thanos.

**Comment by aayvazyan** [comment]
@gracedo @alejandroEsc we have followed your suggestion and tested an umbrella chart which has as subcharts our fork of  [mesosphere/staging/prometheus-operator](https://github.com/mesosphere/charts/tree/master/staging/prometheus-operator) + upstream [`thanos`](https://github.com/banzaicloud/banzai-charts/tree/master/thanos). In this case the deployment of the `prometheus` addon takes **more than 10 minutes** after the `konvoy deploy addons` is finished, this is likely because we don't have the ability to provide `--wait` [flag](https://github.com/mesosphere/kubeaddons/blob/7cbcda6ae4639ee8cbe8e629985404aa9c282a4a/internal/drivers/helm/v2/utils.go#L118) to helm and helm does not control the order of resources deployed in the subcharts.  

IMHO 10 minutes is a significant delay and not acceptable to ship to customers and this is not easy to solve without touching the `prometheus-operator` chart.

Below the umbrella charts for  
- [Prometheus-operator-0.34.x](https://github.com/aayvazyan/charts/tree/tig/prometheus-thanos-v3-0.34.x/staging/prometheus-thanos) 
- [Prometheus-operator-0.35.x](https://github.com/aayvazyan/charts/tree/tig/prometheus-thanos-v3-0.35.x)

Here is the ticket with the issue described and logs attached: https://jira.d2iq.com/browse/D2IQ-65783

The branches with the addons: 
- [kba-0.34.x](https://github.com/aayvazyan/kubernetes-base-addons/blob/tig/prometheus-thanos-v3-0.34.x/addons/prometheus/0.34.x/prometheus-6.yaml#L46)
- [kba-0.35.x](https://github.com/aayvazyan/kubernetes-base-addons/blob/tig/prometheus-thanos-v3-0.35.x/addons/prometheus/0.35.x/prometheus-1.yaml#L46)

So the scenario of having prometheus-operator as a subchart didn't work for us and right now we have **_2 approaches_** ready to go.
1. Having Thanos as a separate addon 
     - [Here the branch with the addon](https://github.com/aayvazyan/kubernetes-base-addons/tree/tig/prometheus-operator-v3-0.34.x)
     - [Here the PR against kubernetes-sre-addons](https://github.com/mesosphere/kubernetes-sre-addons/pull/2) repo
       - Status: Ready and Tested
       - Pros: 
         -  Easy to implement
         - No need to modify our fork of  [mesosphere/staging/prometheus-operator](https://github.com/mesosphere/charts/tree/master/staging/prometheus-operator)
       - Conns:
         - Certain level of expertise needed from operator to configure thanos
         - Thanos and Prometheus become cross-dependant addons.
2. Having Thanos as a subchart in our fork of [mesosphere/staging/prometheus-operator](https://github.com/mesosphere/charts/tree/master/staging/prometheus-operator)
     - Status: Tested on our side it works. 
     - Pros: 
         - Easy to configure, no expertise needed from operator
         - No cross-dependant addons.
     - Conns:
         - Modification of our fork of  [mesosphere/staging/prometheus-operator](https://github.com/mesosphere/charts/tree/master/staging/prometheus-operator) needed.

**Comment by aayvazyan** [comment]
> I think we should pull the current mesosphere `prometheus-operator` (our fork) as a subchart to the parent `prometheus-thanos` chart. So essentially just copying the chart entirely into a `staging/prometheus-thanos/charts/prometheus-operator/` subchart dir. It looks like with these changes, the `prometheus-operator` chart was actually just renamed to `prometheus-thanos` and enhanced to pull `thanos` chart in. But we really want to be customizing our fork of `prometheus-operator` as little as possible - it is still our plan in the future to eventually get rid of our fork entirely and use the official upstream helm chart.
> 
> On that note, we're trying to stay in line with the upstream chart as closely as possible to make that transition smooth. We shouldn't update the subcharts within prometheus-operator since we're still operating off the upstream chart as our base, and each version of the chart has been tested and vetted with each version of the subcharts within it. If we update those subcharts, we can't ensure that those versions are compatible with the version of the prometheus-operator chart.

@gracedo @alejandroEsc these suggestions are pushed, please check and resolve the conversation.

File: staging/prometheus-thanos/crds/crd-alertmanager.yaml (hunk 1 above)
Comment (aayvazyan) [reply staging/prometheus-thanos/crds/crd-alertmanager.yaml]: Resolving this as we are not modifying `prometheus-operator` anymore but have it as a `subchart`.

**Comment by shaneutt** [comment]
It has been some time since there were updates on this PR, is this something we want to close temporarily or is help and review actively needed?

**Comment by aayvazyan** [comment]
A discussion is going on what is the best way to resolve the backward compatibility for combined charts.  
Temporarily closing this until a decision will be made.

**Pull request closed without merging**